### PR TITLE
Remove CoreCLR TargetSpecific test build jobs

### PR DESCRIPTION
**Pull request by sdmaclea**
Refactor targetSpecific test build
Remove targetSpecific build-test-jobs.yml jobs
Build targetSpecific tests in run-test-job.yml jobs

### CODE REVIEW

File: eng/pipelines/common/templates/runtimes/run-test-job.yml [hunk 1]
```diff
@@ -236,6 +238,24 @@ jobs:
...
+            displayName: Install native dependencies
+        - ${{ if eq(parameters.osGroup, 'Windows_NT') }}:
+          # Necessary to install correct cmake version
+          - script: $(Build.SourcesDirectory)\eng\common\init-tools-native.cmd -InstallDirectory $(Build.SourcesDirectory)\native-tools -Force
+            displayName: Install native dependencies
+
+        # Build targetSpecific managed test components
+        - script: $(coreClrRepoRootDir)build-test$(scriptExt) targetSpecific skipnative skipgeneratelayout skiptestwrappers $(buildConfig) $(archType) $(crossArg) $(priorityArg) ci $(librariesOverrideArg)
+          displayName: Build managed test components
+
```
Comment (sdmaclea) [inline eng/pipelines/common/templates/runtimes/run-test-job.yml:257]: The earliest draft of this was also adding an archive of the tests.
I was fiddling with two variants.  
1. Archive all tests
2. Archive target specific tests

I though that if performance was really critical, no artifact would be faster.

**Comment by sdmaclea** [comment]
One potential other problem with this approach is when we run multiple jobs on the same tests.  R2R & !R2R & JitStress* &&&&....

Every job would be building the target specific tests.

**Comment by sdmaclea** [comment]
```
Artifact MicrosoftNetSdkIlPackage_*_checked not found for build
```

I guess the target specific build was generating this too...

**Comment by sdmaclea** [comment]
This seems to be working now.
I am going to push a change to the display names, but this could use at least a preliminary review.

**Comment by hoyosjs** [comment]
The other disadvantage here is retrying a test run will cause a rebuild too. Not sure if this matters too much.

**Comment by safern** [comment]
> The other disadvantage here is retrying a test run will cause a rebuild too. Not sure if this matters too much.

While that is true, I think this will bring faster builds, less legs meaning less prone to infra errors, and less times hitting the network to download artifacts, etc. So as long as the tests are reliable enough, we shouldn't be specializing for retry.

**Comment by sdmaclea** [comment]
There are 5 readytorun tests which are not happy with this change.  I am assuming it is because the tests were `crossgen2` for `x64`.

I assume therefore, these tests need to be marked `ArchSpecific`

File: eng/pipelines/common/templates/runtimes/run-test-job.yml [hunk 2]
```diff
@@ -236,15 +238,35 @@ jobs:
           artifactName: '$(coreClrProductArtifactName)'
           displayName: 'CoreCLR product download for Mono'
 
+
+    # Build target specific tests
+    - ${{ if ne(parameters.corefxTests, true) }}:
```
Comment (trylek) [inline eng/pipelines/common/templates/runtimes/run-test-job.yml:243]: Nit - you should be able to combine the two if statements using the and() function and remove one level of nesting ;-).

**Review by trylek — APPROVED** [review]
Looks great, thank you for all the improvements you're making to our pipelines!

**Comment by sdmaclea** [comment]
I marked the 5 tests failing in p0 target specific..

Running an outerloop p1 run here https://dev.azure.com/dnceng/public/_build/results?buildId=627124&view=results

I am expecting to have to mark some p1 tests too.

**Comment by sdmaclea** [comment]
Closing in favor of #35783

**Pull request closed without merging**